controllers/nutanixcluster_controller.go

```diff
@@ -178,6 +179,13 @@ func (r *NutanixClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	}
 	conditions.MarkTrue(cluster, infrav1.CredentialRefSecretOwnerSetCondition)
 
+	if err := r.reconcileTrustBundleRef(ctx, cluster); err != nil {
+		log.Error(err, fmt.Sprintf("error occurred while reconciling trust bundle ref for cluster %s", capiCluster.Name))
+		conditions.MarkFalse(cluster, infrav1.TrustBundleSecretOwnerSetCondition, infrav1.TrustBundleSecretOwnerSetFailed, capiv1.ConditionSeverityError, err.Error())
+		return reconcile.Result{}, err
+	}
+	conditions.MarkTrue(cluster, infrav1.TrustBundleSecretOwnerSetCondition)
```

Comment:
we should not set this if we never used trust bundle.  err nil is returned in following case as well <pre>
	trustBundleRef := nutanixCluster.GetPrismCentralTrustBundle()
	if trustBundleRef == nil {
		log.Info(fmt.Sprintf("trust bundle ref is nil for cluster %s", nutanixCluster.Name))
		return nil
</pre>

Reply:
good point, changed the logic, although I moved the condition marking logic to the reconcileTrustBundleRef method to keep the logic cleaner.